src/utils/subscription-registry.js

```diff
@@ -220,32 +205,44 @@ class SubscriptionRegistry {
     delayedBroadcasts.sharedMessages += msgString
     const stop = delayedBroadcasts.sharedMessages.length
 
-    // uniqueSendersMap maps from uuid to offset in uniqueSendersVector
+    const uniqueSenders = delayedBroadcasts.uniqueSenders
+    const sharedMessages = delayedBroadcasts.sharedMessages
+
     // each uniqueSender has a vector of "gaps" in relation to sharedMessage
     // senders should not receive what they sent themselves, so a gap is inserted
     // for every send from this sender
-    let pos
     if (sender) {
-      pos = delayedBroadcasts.uniqueSendersMap[sender.uuid]
-      if (pos !== undefined) {
-        delayedBroadcasts.uniqueSendersVector[pos].gaps.push({ start, stop })
-      } else {
-        pos = delayedBroadcasts.uniqueSendersVector.length
-        delayedBroadcasts.uniqueSendersMap[sender.uuid] = pos
-        delayedBroadcasts.uniqueSendersVector[pos] = {
+      const index = utils.sortedIndexBy(uniqueSenders, sender, 'uuid')
+
+      if (uniqueSenders[index] !== sender) {
+        uniqueSenders.splice(index, 0, {
           sender: sender.uuid,
           message: null,
-          gaps: [{ start, stop }]
-        }
+          gaps: []
+        })
       }
+
+      uniqueSenders[index].gaps.push({ start, stop })
     }
 
     // reuse the same timer if already started
     if (!delayedBroadcasts.timer) {
-      if (this._delay !== -1) {
-        delayedBroadcasts.timer = setTimeout(this.onBroadcastTimeout.bind(this), this._delay, delayedBroadcasts)
+      if (this._broadcastTimeout) {
+        delayedBroadcasts.timer = setTimeout(
+          this._onBroadcastTimeout,
+          this._broadcastTimeout,
+          name,
+          uniqueSenders,
+          sharedMessages,
+          sockets
+        )
```

Comment:
In theory this could be replaced with a static `setInterval` dispatcher. However, I'm not sure if the `setTimeout` overhead is significant or not...